test/integration/collection-management/collection.test.ts

```diff
@@ -95,18 +95,14 @@ describe('Collection', function () {
       ]);
     });
 
-    it('should fail due to illegal listCollections', function (done) {
-      expect(() => db.collection(5)).to.throw('Collection name must be a String');
-      expect(() => db.collection('')).to.throw('Collection names cannot be empty');
-      expect(() => db.collection('te$t')).to.throw("Collection names must not contain '$'");
-      expect(() => db.collection('.test')).to.throw(
-        "Collection names must not start or end with '.'"
-      );
-      expect(() => db.collection('test.')).to.throw(
-        "Collection names must not start or end with '.'"
-      );
-      expect(() => db.collection('test..t')).to.throw('Collection names cannot be empty');
-      done();
+    it('fails on server due to invalid namespace', async function () {
+      const error = await db
+        .collection('a\x00b')
+        .insertOne({ a: 1 })
+        .catch(error => error);
+      expect(error['name']).to.equal('MongoServerError');
+      expect(error['code']).to.equal(73);
```

Comment:
You can apply the same chai advice above to all of these. But just some JS knowledge share: `error['code']` and `error.code` are the same thing. Generally, you only ever need to access an object property using computed key syntax (brackets with a string) if the key isn't valid js (ex. `object['key-with-dash']`) or if your key is actually computed (ex. `for (const key of ['a', b']) object[key];`)

Reply:
Good to know this js advice, I'll keep it in mind!